src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -438,8 +446,8 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     completionWork onFailure {
       case e =>
         logger.error(s"Completion work failed for call ${callKey.tag}! " + e.getMessage, e)
-        scheduleTransition(WorkflowFailed)
         self ! PersistenceFailed(callKey, ExecutionStatus.Done)
+        self ! CheckForWorkflowComplete
```

Comment:
I don't believe it can happen in-JVM but if these messages are processed out of order is there any problem?